pkg/fingerprint/engine.go

```
@@ -0,0 +1,804 @@
+	conn, err := e.dialer(dialCtx, "tcp", addr)
+	if err != nil {
+		return false
+	}
+
+	tlsConn := tls.Client(conn, &tls.Config{
+		InsecureSkipVerify: true,
+		ServerName:         t.Host,
+	})
+
+	_ = conn.SetDeadline(time.Now().Add(tlsTimeout))
+	err = tlsConn.Handshake()
```

golangci-lint (2.11.4)

[error] (*crypto/tls.Conn).Handshake must not be called. use (*crypto/tls.Conn).HandshakeContext

(noctx)